src/main/java/org/junit/experimental/results/ResultMatchers.java

```diff
@@ -34,14 +36,21 @@ public ResultMatchers() {
      * Matches if there are {@code count} failures
      */
     public static Matcher<PrintableResult> failureCountIs(final int count) {
+        return failureCount(equalTo(count));
+    }
+
+    /**
+     * Matches if the number of failures matches {@code countMatcher}
+     */
+    public static Matcher<PrintableResult> failureCount(final Matcher<Integer> countMatcher) {
         return new TypeSafeMatcher<PrintableResult>() {
             public void describeTo(Description description) {
-                description.appendText("has " + count + " failures");
+                description.appendText("has a number of failures matching " + countMatcher);
```

Comment:
I understand your concern but the original string doesn't work well on failure with greaterThan and similar:

- Expected: has a value equal to or greater than <2> failures
- Expected: has a value greater than <2> failures
- Expected: has a value less than <1> failures

Sounds better:

    Expected: has a number of failures matching a value equal to or greater than <2>

But we could have the original failureCountIs override describeTo of the new overloaded version..